pkg/controller/common/util.go

```diff
@@ -434,38 +437,97 @@ func GetVolumeMode(pvc *corev1.PersistentVolumeClaim) corev1.PersistentVolumeMod
 	return util.ResolveVolumeMode(pvc.Spec.VolumeMode)
 }
 
-// GetStorageClassByName looks up the storage class based on the name. If no storage class is found returns nil
+// GetStorageClassByName looks up the storage class based on the name
+// If name is nil, it looks for the default k8s storage class storageclass.kubernetes.io/is-default-class
+// If no storage class is found, returns nil
 func GetStorageClassByName(ctx context.Context, client client.Client, name *string) (*storagev1.StorageClass, error) {
+	if name == nil {
+		return GetFallbackStorageClass(ctx, client, false)
+	}
+
 	// look up storage class by name
-	if name != nil {
-		storageClass := &storagev1.StorageClass{}
-		if err := client.Get(ctx, types.NamespacedName{Name: *name}, storageClass); err != nil {
-			if k8serrors.IsNotFound(err) {
-				return nil, nil
-			}
-			klog.V(3).Info("Unable to retrieve storage class", "storage class name", *name)
-			return nil, errors.Errorf("unable to retrieve storage class %s", *name)
+	storageClass := &storagev1.StorageClass{}
+	if err := client.Get(ctx, types.NamespacedName{Name: *name}, storageClass); err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil, nil
+		}
+		klog.V(3).Info("Unable to retrieve storage class", "storage class name", *name)
+		return nil, errors.Errorf("unable to retrieve storage class %s", *name)
+	}
+
+	return storageClass, nil
+}
+
+// GetStorageClassByNameWithVirtFallback looks up the storage class based on the name
+// If name is nil, it looks for the following, in this order:
+// default kubevirt storage class (if the caller is interested) storageclass.kubevirt.io/is-default-class
+// default k8s storage class storageclass.kubernetes.io/is-default-class
+// If no storage class is found, returns nil
+func GetStorageClassByNameWithVirtFallback(ctx context.Context, client client.Client, name *string, considerVirtSc bool) (*storagev1.StorageClass, error) {
```

Comment:
Why both `GetStorageClassByName` and `GetStorageClassByNameWithVirtFallback`?  I'm sure there's a good reason.  Prob worth adding a comment why 

Also, bool params should be avoided when possible.  Maybe it would be better to have function take a DataVolume as a param instead?

Reply:
> Why both `GetStorageClassByName` and `GetStorageClassByNameWithVirtFallback`? I'm sure there's a good reason. Prob worth adding a comment why

Renamed to better communicate the reason.
Basically, there will still be flows where you don't want to take content type into account, quick example is when using the `PVC` API

> Also, bool params should be avoided when possible. Maybe it would be better to have function take a DataVolume as a param instead?

Yes definitely. Hopefully this is better now, whole DV was a bit overkill but content type should be fine